nomad/acl_endpoint.go

```diff
@@ -396,6 +398,16 @@ func (a *ACL) Bootstrap(args *structs.ACLTokenBootstrapRequest, reply *structs.A
 		Global:     true,
 		CreateTime: time.Now().UTC(),
 	}
+
+	// if a token has been passed in from the API overwrite the generated one.
+	if providedTokenID != "" {
+		if helper.IsUUID(providedTokenID) {
+			args.Token.SecretID = providedTokenID
+		} else {
+			return structs.NewErrRPCCodedf(400, "invalid acl token: %v", err)
```

Comment:
`err` is always `nil` at this point because our conditional check for UUID doesn't return an error:

```suggestion
			return structs.NewErrRPCCodedf(400, "invalid acl token")
```